Modules/Microsoft365DSC/DSCResources/MSFT_AADPasswordRuleSettings/MSFT_AADPasswordRuleSettings.psm1

```diff
@@ -0,0 +1,483 @@
+            }
+
+            Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-M365DscHashtableToString -Hashtable $result)"
+            return $result
+        }
+    }
+    catch
+    {
+        New-M365DSCLogEntry -Message 'Error retrieving data:' `
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullReturn
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        [ValidateSet('Yes')]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.UInt32]
+        $LockoutThreshold,
+
+        [Parameter()]
+        [System.UInt32]
+        $LockoutDurationInSeconds,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableBannedPasswordCheck,
+
+        [Parameter()]
+        [System.String[]]
+        $BannedPasswordList,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnableBannedPasswordCheckOnPremises,
+
+        [Parameter()]
+        [validateset('Enforced', 'Audit')]
+        [System.String]
+        $BannedPasswordCheckOnPremisesMode,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $ApplicationSecret,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    Write-Verbose -Message 'Setting configuration of Azure AD Groups Settings'
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $currentPolicy = Get-TargetResource @PSBoundParameters
+
+    # Policy should exist but it doesn't
+    $needToUpdate = $false
+    if ($Ensure -eq 'Present' -and $currentPolicy.Ensure -eq 'Absent')
+    {
+        #$template = Get-MgBetaDirectorySettingTemplate -All | Where-Object -FilterScript {$_.Displayname -eq 'Password Rule Settings'}
+        $Policy = New-MgBetaDirectorySetting -TemplateId '5cf42378-d67d-4f36-ba46-e8b86229381d' | Out-Null
```

Comment:
I see the above line as commented. Is there a way to dynamically retrieve this instead of hardcoding the template's id?

Reply:
The original solution was to get the template filtered by displayname and use the returned id. I was in a hurry to go on vacation and didn't have time to 'fix' the problem of the cmdlet being rejected as unknown by the unit-test kicked off by the PR.